### PR TITLE
Feature: Start flow APIs

### PR DESCRIPTION
**Pull request by anshulrr**
1. Start flow for contact API
2. Start flow for contacts of a group API

### CODE REVIEW

File: api.docs/includes/_flows.md [hunk 1]
```diff
@@ -419,6 +419,123 @@ Type | Description
...
+  	success
+  	errors {
+    	key
+  		message
+  	}
+  }
+}
+
+{
+  "id": "1",
```
Comment (pankaj-ag) [inline api.docs/includes/_flows.md]: I think this should be flow-id, ID might be confusing

File: api.docs/includes/_flows.md [hunk 2]
```diff
@@ -419,6 +419,123 @@ Type | Description
...
```
Comment (pankaj-ag) [inline api.docs/includes/_flows.md]: Same here

File: lib/glific_web/resolvers/flows.ex [hunk 3]
```diff
@@ -74,4 +76,30 @@ defmodule GlificWeb.Resolvers.Flows do
...
     end
   end
+
+  @doc false
+  @spec start_contact_flow(Absinthe.Resolution.t(), %{id: integer, contact_id: integer}, %{
+          context: map()
+        }) ::
+          {:ok, any} | {:error, any}
+  def start_contact_flow(_, %{id: id, contact_id: contact_id}, _) do
+    with {:ok, flow} <- Repo.fetch(Flow, id),
```
Comment (pankaj-ag) [inline lib/glific_web/resolvers/flows.ex]: Why do we need to fetch the flow here? can't you just pass this to context and get_Cached_flow will load it.

File: lib/glific_web/resolvers/flows.ex [hunk 4]
```diff
@@ -74,4 +76,30 @@ defmodule GlificWeb.Resolvers.Flows do
...
+    end
+  end
+
+  @doc false
+  @spec start_group_flow(Absinthe.Resolution.t(), %{id: integer, group_id: integer}, %{
+          context: map()
+        }) ::
+          {:ok, any} | {:error, any}
+  def start_group_flow(_, %{id: id, group_id: group_id}, _) do
+    with {:ok, flow} <- Repo.fetch(Flow, id),
```
Comment (pankaj-ag) [inline lib/glific_web/resolvers/flows.ex]: Same here

File: lib/glific/flows.ex [hunk 5]
```diff
@@ -319,4 +322,38 @@ defmodule Glific.Flows do
...
+      {:error, ["contact", "Cannot send the message to the contact."]}
+    end
+  end
+
+  @doc """
+  Start flow for contacts of a group
+  """
+  @spec start_group_flow(Flow.t(), Group.t()) :: {:ok, Flow.t()}
+  def start_group_flow(%Flow{} = flow, %Group{} = group) do
+    {:ok, flow} = get_cached_flow(flow.id, %{id: flow.id})
```
Comment (pankaj-ag) [inline lib/glific/flows.ex:346]: Let's not prefetch the flow because get_cached_flow will fetch it in any case.

**Review by pankaj-ag — CHANGES_REQUESTED** [review]

File: lib/glific_web/resolvers/flows.ex (hunk 3 above)
Comment (anshulrr) [reply lib/glific_web/resolvers/flows.ex]: get_cached_flow doesn't have a check for incorrect data, we need to fix it to make it work

**Review by pankaj-ag — APPROVED** [review]

**Pull request merged**